engine/src/main/java/org/pentaho/di/trans/steps/csvinput/CsvInput.java

```diff
@@ -1161,4 +1162,15 @@ public boolean init( StepMetaInterface smi, StepDataInterface sdi ) {
 
     return strings.toArray( new String[ strings.size() ] );
   }
+
+  String filenameValidatorForInputFiles( String filename ) {
+    if ( getTransMeta().getRepository() != null && getTransMeta().getRepository().isConnected() && filename
```

Comment:
Repetitively calling getTransMeta() and getRepository(). For one row, maybe insignificant, but for a billion rows, very significant. This should read:

... snip ...
Repository aRepo = getTransMeta().getRepository();
if ( aRepo != null && aRepo.isConnected() && filename.contains( Const.INTERNAL_VARIABLE_ENTRY_CURRENT_DIRECTORY ) ) {
... snip ...

Reply:
Totally agree. Good catch.
I'm closing this PR until we have some feedback from Jens, if we should use this variable when running transformations in CDA. That "replace fix" is an ugly one - we want first to ensure that we support this behavior with our new variable Internal.Entry.Current.Directory.